packages/openapi3/src/openapi.ts

```diff
@@ -225,6 +222,28 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
     tags = new Set();
   }
 
+  function resolveServers(servers: http.HttpServer[]): OpenAPI3Server[] {
+    return servers.map((server) => {
+      const variables: Record<string, OpenAPI3ServerVariable> = {};
```

Comment:
Not sure if we do this when constructing the HttpServer, but we should have validation that each of the parameter types in server variables is a string type (string literal, string, enum with string value, union with all string options)

Reply:
So I know this is openapi restriction but feel like we might want to allow other types. I guess we could start with more limitation and relax later

Reply:
nvm didn't noticed the comment was on the openapi lib